xbmc/guilib/GUIFontTTF.cpp

```diff
@@ -870,9 +875,10 @@ bool CGUIFontTTF::CacheCharacter(wchar_t letter, uint32_t style, Character* ch,
 
     // check we have enough room for the character.
     // cast-fest is here to avoid warnings due to freeetype version differences (signedness of width).
-    if (static_cast<int>(m_posX + bitGlyph->left + bitmap.width) > static_cast<int>(m_textureWidth))
+    if (static_cast<int>(m_posX + bitGlyph->left + bitmap.width +
+                         SPACING_BETWEEN_CHARACTERS_IN_TEXTURE) > static_cast<int>(m_textureWidth))
     { // no space - gotta drop to the next line (which means creating a new texture and copying it across)
-      m_posX = 0;
+      m_posX = 1;
```

Comment:
What's the reason for the `1` here? Extra margin like in https://github.com/xbmc/xbmc/pull/21511?

Reply:
You don't want to have the glyphs touching the border. You might get some rogue (non-zero) texels out of the neighboring border region.

Reply:
I think that the previous version didn't prevent this case. However, the odds of having issues due to "touching" glyphs was astronomical high, due to the generous padding.

And even so, this only shows when not sampling from the middle of the texels, which is not often the case with our GUI.